pandas/tests/scalar/test_timestamp.py

```diff
@@ -695,6 +695,27 @@ def test_round(self):
         expected = Timestamp('20130101')
         assert result == expected
 
+        # GH 19206
+        dt = Timestamp('2117-01-01 00:00:45')
+        result = dt.floor('15s')
+        expected = Timestamp('2117-01-01 00:00:45')
+        assert result == expected
+
+        dt = Timestamp('2117-01-01 00:00:45.000000012')
+        result = dt.floor('10ns')
+        expected = Timestamp('2117-01-01 00:00:45.000000010')
+        assert result == expected
+
+        dt = Timestamp('1823-01-01 00:00:01')
+        result = dt.floor('1s')
+        expected = Timestamp('1823-01-01 00:00:01')
+        assert result == expected
+
+        dt = Timestamp('1823-01-01 00:00:01.000000012')
+        result = dt.floor('10ns')
+        expected = Timestamp('1823-01-01 00:00:01.000000010')
+        assert result == expected
+
         # ceil
```

Comment:
can we test for ceil as well. If you can parametrize this all the better (if not ok too).

Also ok with splitting this giant test to test_round, test_floor, test_ceil

Reply:
Still working on other change requests, but just wanted to get feedback on refactoring for timestamp tests. There are so many tests that it seemed like a mixed approach of splitting tests and so that failed assertions could be easily traced, and parametrizing to consolidate common code was the way to go.

Reply:
see below. we *need* to parameterize as much as possible to cover the cases. and you can still run them individually as well (optionally you can pass ``ids`` to create them)